js/quote-generator.js

```diff
@@ -0,0 +1,55 @@
+const randomBtn = document.querySelector("#random-btn");
+const allAuthorsBtn = document.querySelector("#all-authors-btn");
+const quote = document.querySelector(".quote");
+let quoteAuthor = document.querySelector(".quote-author");
+let quoteGenre = document.querySelector(".quote-genre");
+const allAuthorsContainer = document.querySelector(".all-authors-container");
+const gridContainer = document.querySelector(".grid-container");
+const topButtons = document.querySelector(".top-buttons");
+const footer = document.querySelector("footer");
+
+async function getRandomQuote() {
+  const promise = await fetch("https://api.quotable.io/random");
+  const result = await promise.json();
+  quote.textContent = result.content;
+  quoteAuthor.textContent = result.author;
+  quoteGenre.textContent = result.tags.join(", ");
+}
+
+randomBtn.addEventListener("click", getRandomQuote);
+getRandomQuote();
+
+async function getAllAuthors() {
+  const promise = await fetch(
+    `https://api.quotable.io/authors?sortBy=quoteCount&order=desc&limit=10`
+  );
+  const result = await promise.json();
+  let count = 1;
+  for (let oneAuthor of result.results) {
+    const authorContainer = document.createElement("div");
+    authorContainer.classList.add("all-authors-item");
+    const authorInfo = document.createElement("h3");
+    authorInfo.classList.add("all-authors-item-content");
+    authorInfo.textContent = `${count}. ${oneAuthor.name} has ${oneAuthor.quoteCount} great qoutes`;
+    count++;
+    authorContainer.appendChild(authorInfo);
+    allAuthorsContainer.appendChild(authorContainer);
+  }
+}
+
+allAuthorsBtn.addEventListener("click", getAllAuthors(), { once: true });
```

Comment:
User sees list of authors only when he visits the proper page. But since you call this method immediately, you load all of the authors even when user haven't checked the proper page yet. This is not optimised. 

Call your method only when user visits you page. Don't forget to handle loading & error states of your HTTP request